front/src/views/user/add-edit-route-form/add-edit-route-form.view.js

```diff
@@ -0,0 +1,149 @@
+          this.fileSizeError = true;
+        }
+      }
+    },
+    onPickFile () {
+      this.$refs.fileInput.click();
+    },
+    deleteFile (fileName) {
+      this.$store.dispatch('waypoints/deleteFile', [fileName, +this.waypointId]);
+    },
+    createWaypoint () {
+      this.$refs.form.validate();
+      if (!this.formValidity) {
+        return;
+      }
+      this.sendBtnDisable = true;
+      const { departureTime, arrivalTime, ...rest } = this.newWaypointValues;
+      this.$store.dispatch(
+        this.isEditForm ? 'waypoints/updateWaypoint' : 'waypoints/insertWaypoint', {
+          ...rest,
+          departureDate: zonedTimeToUtc(
+            `${this.newWaypointValues.departureDate}T${this.newWaypointValues.departureTime}`),
+          arrivalDate: zonedTimeToUtc(
+            `${this.newWaypointValues.arrivalDate}T${this.newWaypointValues.arrivalTime}`),
+          pathLength: +this.newWaypointValues.pathLength,
+          tripId: +this.id,
+          newId: +this.waypointId || 0
+        })
+        .then((r) => {
+          if (this.tempFiles.length === 0 || this.isEditForm) {
+            return Promise.resolve();
+          }
+          const formData = new FormData();
+          this.tempFiles.forEach(file => {
+            formData.append('files', file);
+          });
+
+          return this.$store.dispatch('waypoints/sendMultipleFiles', [formData, r.data]);
+        })
+        .then(() => this.$store.dispatch('waypoints/loadWaypoints', [+this.id, true]))
+        .then(() => this.$router.push({ name: 'MyHistoryFututeRoute' }));
+    }
+  },
+  created () {
+    if (this.isEditForm) {
+      this.$store.dispatch('waypoints/loadWaypoints', [ +this.id ])
+        .then(() => {
+          const waypoints = this.$store.state.waypoints.waypoints,
+                currentWaypoint = waypoints.find(waypoint => waypoint.id === +this.waypointId),
+                nextWaypoint
+                 = waypoints[waypoints.findIndex(waypoint => waypoint === currentWaypoint) + 1];
+
+          this.newWaypointValues.departureCity = currentWaypoint.city;
+          this.newWaypointValues.departureDate
+          = format(new Date(currentWaypoint.departureDate), 'yyyy-MM-dd');
+          this.newWaypointValues.departureTime
+          = format(new Date(currentWaypoint.departureDate), 'HH:mm');
+
+          this.newWaypointValues.arrivalCity = nextWaypoint.city;
+          this.newWaypointValues.arrivalDate
+          = format(new Date(currentWaypoint.arrivalDate), 'yyyy-MM-dd');
+          this.newWaypointValues.arrivalTime
+          = format(new Date(currentWaypoint.arrivalDate), 'HH:mm');
+
+          this.newWaypointValues.pathLength = currentWaypoint.pathLength;
+          this.newWaypointValues.pathTime
+          = `${currentWaypoint.pathTime.hours}:${currentWaypoint.pathTime.minutes}`;
+          this.newWaypointValues.details = currentWaypoint.details;
+          this.newWaypointValues.transport = currentWaypoint.transport;
```

Comment:
do we need it yet?

Reply:
yes, because date(time)Pickers works with strings, formatted like this: "yyyy-MM-dd'" ("HH:mm")